### ISSUE TITLE
Adding EMS sensor objects to an IDF in EPPY

### ISSUE THREAD

**Issue by amr-auf** [santoshphilip/eppy#383]
Hi,

Thank you all for the amazing library. It is helping a lot!

I was trying to use Eppy to write EnergyManagementSystem idf objects. I used Eppy to declare global and internal variables, and they worked. 
However, I could not add EMS sensors using the same workflow. I think the problem is attributed to the field names for a sensor object. 

The EMS sensor name and field entries should be okay as I tried them in DesignBuilder and EnergyPlus before trying to use them in Eppy.

You can find below two attempts for the code, and both gave an error that the field could not be found in the IDD. The difference between the attempts is replacing ":" with "_". Spaces are replaced with "_" in both attempts.

**Attempt 1:**

for zone in idf.idfobjects["ZONE"]: 
        idf.newidfobject("ENERGYMANAGEMENTSYSTEM:SENSOR")
        idf.idfobjects["ENERGYMANAGEMENTSYSTEM:SENSOR"][-1].Name = f"InfiltrationACH_{zone.Name}"
        idf.idfobjects["ENERGYMANAGEMENTSYSTEM:SENSOR"][-1].Output_Variable_or_Output_Meter_Index_Key_Name = f"{zone.Name}"
        idf.idfobjects["ENERGYMANAGEMENTSYSTEM:SENSOR"][-1].Output_Variable_or_Output_Meter_Name = "Zone Infiltration Air Change Rate"

**Attempt 2:**
for zone in idf.idfobjects["ZONE"]: 
        idf.newidfobject("ENERGYMANAGEMENTSYSTEM:SENSOR")
        idf.idfobjects["ENERGYMANAGEMENTSYSTEM:SENSOR"][-1].Name = f"InfiltrationACH_{zone.Name}"
        idf.idfobjects["ENERGYMANAGEMENTSYSTEM:SENSOR"][-1].Output:Variable_or_Output:Meter_Index_Key_Name = f"{zone.Name}"
        idf.idfobjects["ENERGYMANAGEMENTSYSTEM:SENSOR"][-1].Output:Variable_or_Output:Meter_Name = "Zone Infiltration Air Change Rate"

Thank you very much,
Amr

**Comment by amr-auf** [comment]
I found out that I can access the field names for an IDF object using the .fieldnames.
The correct field names for EMS sensors in Eppy should be

['key',
 'Name',
 'OutputVariable_or_OutputMeter_Index_Key_Name',
 'OutputVariable_or_OutputMeter_Name']
 
 Thank you.

**Comment by santoshphilip** [comment]
@amr-auf , Is everything working now ?

**Comment by amr-auf** [comment]
Yes, Santosh. Everything is good now. Thank you very much!

**Comment by santoshphilip** [comment]
How did you find the function .fieldnames().
It is a very useful function. I am sorry you did not know of it earlier.
I am trying to figure out where to put in in the documentation. Any thoughts ? Should it be in the tutorial ?

**Comment by amr-auf** [comment]
Thank you Santosh for your attention here. I think the tutorials are great and very clear. They cover the basics of Eppy in a clear step by step approach. I was new to Eppy (and even to EnergyPlus), and the tutorials were enough to give me the confidence to get started.

You might just add a line that explains that if the field name of an IDF object includes a colon ( : ), this colon should be omitted. Just similar to the way you mentioned that all spaces between field names should be replaced with underscores ( _ ). 

Using .fieldnames( ) is also a great way to know the available field names for an IDF object. I knew about it while searching the discussions you had in https://github.com/santoshphilip/eppy/issues/341. I found the .fieldnames( ) very useful. Before knowing it, I had to use the EP-IDF editor to know all the available fields for a certain IDF object.

Many thanks
